MoltenVK/MoltenVK/GPUObjects/MVKSwapchain.mm

```diff
@@ -582,6 +582,30 @@ static CALayerContentsGravity getCALayerContentsGravity(VkSwapchainPresentScalin
 	_presentHistoryIndex = (_presentHistoryIndex + 1) % kMaxPresentationHistory;
 }
 
+void MVKSwapchain::setLayerNeedsDisplay(const VkPresentRegionKHR* pRegion) {
+	if (!pRegion || pRegion->rectangleCount == 0) {
+		[_mtlLayer setNeedsDisplay];
```

Comment:
My first instinct was to muse aloud, "If we didn't call `-[CALayer setNeedsDisplay]` before, how did it get updated at all in the first place?"

Then I realized that it would be easier to make my case if I had some hard numbers. This is what I found. For reference, I used a MacBook Pro with M1 Pro running the cube demo with the `IMMEDIATE` present mode on macOS Sonoma beta 3 for these measurements:

```
Baseline:
[mvk-info]   Frame interval avg: 8.300 ms, latest: 7.,829 ms, min: 1.287 ms, max: 14.037 ms, count: 4722
With incremental_present support:
[mvk-info]   Frame interval avg: 8.302 ms, latest: 7.840 ms, min: 1.064 ms, max: 14.662 ms, count: 11557
With incremental_present enabled:
[mvk-info]   Frame interval avg: 8.292 ms, latest: 8.320 ms, min: 1.234 ms, max: 15.378 ms, count: 11354
```

"Baseline" is for MoltenVK without my change. "With `incremental_present` support" is with my change. In the "With `incremental_present` enabled" case, I additionally gave the `--incremental_present` option to the demo. As you can see, the differences between all three cases are negligible and well within random jitter.

Reply:
My point is, I actually think that `-[CAMetalDrawable present]` is calling `-[CALayer setNeedsDisplay]` for us. My hope was that it would skip that call if the layer were already marked as needing update.

Reply:
> I actually think that -[CAMetalDrawable present] is calling -[CALayer setNeedsDisplay] for us. 

Yes. That's what I assumed too. Or possibly something internal and more efficient, since it's unlikely `CAMetalLayer` needs to trigger things like `drawLayer:inContext:` in the view.

Thanks for doing the comparison testing. Looks like it's not causing any trouble then. Too bad it maxed at 120Hz, as I can't imagine the cube itself takes anything close to 8 ms to render.